### PR TITLE
Enhancement: Multi-merging in branch view

### LINKED ISSUE #420: Multi-merging in branch view
The <kbd>m</kbd> key states "merge selected into active branch", but it only does one at a time.
I would like to see the ability to select multiple (either text selection or multiple cursors if possible) branches to merge from.

### PR DESCRIPTION
**Pull request by stoivo**
Fix #420

This make it work for multiple selections.
This can be added for multiple lines selections as well, when this will be done
I think we should edit `common/util/view.py@get_lines_from_regions`.

### CODE REVIEW

**Comment by stoivo** [comment]
@divmain, does this looks ok to you?

**Comment by stoivo** [comment]
Can this we merged?

**Comment by divmain** [comment]
Sorry @stoivo, somehow this went off my radar!

File: core/interfaces/branch.py [hunk 1]
```diff
@@ -486,26 +486,31 @@ def run(self, edit):
...
-        if not line:
-            return
+        branches = list(self.selected_branches(sel) for sel in self.view.sel())
+        current_branch_name = self.get_current_branch_name()
+
+        branches = set(branch for branch in branches if branch and not branch == current_branch_name)
+        self.merge(branches)
+        util.view.refresh_gitsavvy(self.view)
+
+    def selected_branches(self, sel):
```
Comment (divmain) [inline core/interfaces/branch.py]: Would prefer to rename `get_selected_branches`.

File: core/interfaces/branch.py [hunk 2]
```diff
@@ -486,26 +486,31 @@ def run(self, edit):
...
+        lines = util.view.get_lines_from_regions(self.view, [sel])
+        return self.branch_extraction(sel, line) for line in lines
 
+    def branch_extraction(self, selection, line):
```
Comment (divmain) [inline core/interfaces/branch.py]: Would prefer rename to `get_selected_branch_name`.

**Comment by divmain** [comment]
Very minor comments, once those are changed feel free to merge without a re-review.

**Comment by stoivo** [comment]
Fixed comments.

**Pull request merged**